lib/portage/package/ebuild/doebuild.py

```diff
@@ -2560,17 +2552,51 @@ def _post_src_install_write_metadata(settings):
                 except OSError:
                     pass
                 continue
-            with open(
-                _unicode_encode(
-                    os.path.join(build_info_dir, k),
-                    encoding=_encodings["fs"],
-                    errors="strict",
-                ),
-                mode="w",
-                encoding=_encodings["repo.content"],
+
+            metadata_buffer[k] = v
+
+    # We could skip this for non-binpkgs but there doesn't seem to be much
+    # value in that, as users shouldn't downgrade libc anyway.
+    #
+    # TODO: We could skip the injection for packages installing no ELF files
```

Comment:
I'm going to see about doing this (the conditional part, not the "start doing everything in Python" bit), although I'm worried it may be messier than the PR currently is.